### PR TITLE
Restructure & cleanup

### PR DESCRIPTION
**Pull request by danielocfb**
Please see individual commits for descriptions.

### CODE REVIEW

File: src/elf/parser.rs [hunk 1]
```diff
@@ -88,7 +88,7 @@ fn get_elf_section_name<'a>(sect: &Elf64_Shdr, strtab: &'a [u8]) -> Option<&'a s
     extract_string(strtab, sect.sh_name as usize)
 }
 
-struct Elf64ParserBack {
+struct ElfParserBack {
     ehdr: Option<Elf64_Ehdr>,
     shdrs: Option<Vec<Elf64_Shdr>>,
```
Comment (anakryiko) [inline src/elf/parser.rs:92]: I think it does deal only with 64-bit ELF files, which is why it's called Elf64Parser. In the future we can start supporting 32-bit ELFs, so maybe let's keep it as is? Unless the idea is to make ElfParser to work with both 32-bit and 64-bit files transparently?

File: src/elf/types.rs [hunk 2]
```diff
@@ -1,178 +1,41 @@
...
-}
+pub use libc::Elf64_Addr;
+pub use libc::Elf64_Half;
+pub use libc::Elf64_Off;
+pub use libc::Elf64_Phdr;
+pub use libc::Elf64_Shdr;
+pub use libc::Elf64_Sxword;
+pub use libc::Elf64_Sym;
+pub use libc::Elf64_Word;
+pub use libc::Elf64_Xword;
```
Comment (anakryiko) [inline src/elf/types.rs:9]: these types are supposed to be part of libelf, not libc. Is it coincidental that we get those exposed from libc as well?

File: src/elf/parser.rs (hunk 1 above)
Comment (danielocfb) [reply src/elf/parser.rs:92]: I think we should try and handle it transparently, certainly if the intention is to expose that type.

File: src/elf/types.rs (hunk 2 above)
Comment (danielocfb) [reply src/elf/types.rs:9]: I understand they are not part of libc the system library. But they are part of `libc` the crate: https://docs.rs/libc/0.2.139/src/libc/unix/linux_like/linux/mod.rs.html

That is by design to the degree I can tell.

Comment (anakryiko) [reply src/elf/types.rs:9]: ah, I see, libc is not just libc :) sounds good then

File: src/elf/parser.rs (hunk 1 above)
Comment (anakryiko) [reply src/elf/parser.rs:92]: Yeah, I think transparently would be easier to develop and maintain, as parts that are not 32/64-specific would be more naturally contained.

**Review by anakryiko — APPROVED** [review]

**Pull request merged**